src/app/test/fakes/PageInfo.ts

```diff
@@ -1,14 +1,20 @@
+import { RenderMode } from "@angular/ssr";
 import { homeCategory, homeMenuItem } from "@components/home/home.menus";
 import { IPageInfo } from "@helpers/page/pageInfo";
 import { modelData } from "@test/helpers/faker";
 
-export function generatePageInfo(data?: Partial<IPageInfo>): IPageInfo {
+export function generatePageInfo(data?: Partial<IPageInfo>): Required<IPageInfo> {
   return {
     fullscreen: modelData.datatype.boolean(),
     category: homeCategory,
     pageRoute: homeMenuItem,
     resolvers: {},
     menus: {},
+    renderMode: modelData.helpers.arrayElement([
+      RenderMode.Server,
+      RenderMode.Client,
+      RenderMode.Prerender,
+    ]),
```

Comment:
probably don't want to randomize this, it could create flaky tests that render different ways? It depends how often we use this pageinfo generator/helper...

Reply:
We use the `PageInfo` generator/helper a lot.

Most of our Testbed's don't actually have a router, and those that do provide their own router, meaning that the mocked routes don't use `StrongRoute`'s or `PageInfo` at all.

e.g.
https://github.com/QutEcoacoustics/workbench-client/blob/bfee1a82c92a8f56d2b192dbb42cdb55191c1545/src/app/directives/strongRoute/strong-route-active.directive.spec.ts#L38-L48

Additionally, all of our tests are client-side only.\
We don't have any ssr or e2e tests.
Meaning that it's not possible to provide `ServerRoute`'s (which is the only type of router that takes a `RenderMode`).

Reply:
![Uploading image.png…]()